src/endpoints/admin.rs

```diff
@@ -19,22 +23,26 @@ struct AdminTemplate<'a> {
     update: &'a Option<Version>,
 }
 
-#[get("/admin")]
-pub async fn get_admin() -> Result<HttpResponse, Error> {
-    return Ok(HttpResponse::Found()
-        .append_header(("Location", format!("{}/auth_admin", ARGS.public_path_as_str())))
-        .finish());
+pub async fn get_admin() -> Result<impl IntoResponse, AppError> {
+    Ok((
+        StatusCode::OK,
+        [(
+            header::LOCATION,
+            format!("{}/auth_admin", ARGS.public_path_as_str()),
+        )],
+        "".to_string(),
+    ))
 }
 
-#[post("/admin")]
+#[axum::debug_handler]
```

Comment:
why is this annotated with debug_handler?

Reply:
Right. We can remove it again. Axum is quite awesome if you find a non explanatory error why you can't use an endpoint in a handler just add debug_handler and you get a clear and understandable explanation. In this case this was the addition that you can't hold a mutex across await points